dlt/common/configuration/providers/google_secrets.py

```diff
@@ -1,13 +1,36 @@
 import base64
+import string
+import re
+from typing import Tuple
 
 from dlt.common import json
 from dlt.common.configuration.specs import GcpServiceAccountCredentials
 from dlt.common.exceptions import MissingDependencyException
-
 from .toml import VaultTomlProvider
 from .provider import get_key_name
 
 
+def strip_punctuation(in_string: str) -> str:
+    """Replace punctuation characters in a string
+
+    Note: We exclude `_` and `-` from punctuation characters
+
+    Args:
+        in_string(str): input string
+
+    Returns:
+        (str): a string without punctuatio characters and whitespaces
+    """
+    whitespace = re.compile("\s+")
+    punctuation_chars = set(string.punctuation) - {"-", "_"}
+    for char in in_string:
+        if char in punctuation_chars:
+            in_string = in_string.replace(char, "")
```

Comment:
Most likely can be replaces with some pre-compiled regex that will replace all punctuation chars from `in_string`.

Reply:
Yep you are right regex suits better